kombu/messaging.py

```diff
@@ -173,14 +173,43 @@ def publish(self, body, routing_key=None, delivery_mode=None,
                 # XXX declare should be a Set.
                 declare.append(self.exchange)
 
+        old_transport_opts = None
         if retry:
             _publish = self.connection.ensure(self, _publish, **retry_policy)
-        return _publish(
+
+            # If transport options are not provided for the connection
```

Comment:
Perhaps the reasoning here, could be transferred to the commit message instead.

Reply:
They could but the code isn't clear without them.